tests/test_integration.py

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/env python3
+# Copyright (c) Facebook, Inc. and its affiliates.
+# All rights reserved.
+
+# This source code is licensed under the license found in the
+# LICENSE file in the root directory of this source tree.
+
+"""
+These tests check that the server can handle different experiments
+(multi/single stimuli, multi/single outcome). They ensure that the
+data is correctly stored in the database tables (raw, param, and outcome).
+It also checks that the experiment table is correctly populated
+(generate_experiment_table method).
+"""
+
+import json
+import logging
+import unittest
+import uuid
+from unittest.mock import call, MagicMock, patch, PropertyMock
+from parameterized import parameterized
+from itertools import product
+
+import aepsych.server as server
+import aepsych.utils_logging as utils_logging
+import torch
+from aepsych.config import Config
+
+params = {'singleStimuli':{
+    'x1':[0.1, 0.2, 0.3, 1, 2, 3, 4],
+    'x2':[4, 0.1, 3, 0.2, 2, 1, 0.3]
+}, 'multiStimuli':{
+    'x1': [[0.1, 0.2], [0.3, 1], [2, 3], [4, 0.1], [0.2, 2], [1, 0.3], [0.3, 0.1]],
+    'x2': [[4, 0.1], [3, 0.2], [2, 1], [0.3, 0.2], [2, 0.3], [1, 0.1], [0.3, 4]]
+}}
+
+outcomes = {'singleOutcome':[1, -1, 0.1, 0, -0.1, 0, 0],
+            'multiOutcome':[[1, 0], [-1, 0], [0.1, 0], [0, 0], [-0.1, 0], [0, 0], [0, 0]]}
+
+all_tests = ['singleStimuli_singleOutcome',
```

Comment:
Minor nit -- if you just make the tests tuples of `stimuli, outcome` objects (which you can generate from an `itertools.product(params, outcomes)` then you don't have to construct these strings and then parse them later.